server/athenian/api/models/web/filtered_code_check_run.py

```diff
@@ -0,0 +1,188 @@
+from datetime import datetime
+from typing import List, Optional
+
+from athenian.api.models.web.base_model_ import Model
+from athenian.api.models.web.code_check_run_statistics import CodeCheckRunStatistics
+
+
+class FilteredCodeCheckRun(Model):
+    """Mined information about a code check run."""
+
+    openapi_types = {
+        "title": str,
+        "repository": str,
+        "last_execution": datetime,
+        "total_stats": CodeCheckRunStatistics,
+        "prs_stats": CodeCheckRunStatistics,
+        "size_groups": List[int],
+    }
+
+    attribute_map = {
+        "title": "title",
+        "repository": "repository",
+        "last_execution": "last_execution",
+        "total_stats": "total_stats",
+        "prs_stats": "prs_stats",
+        "size_groups": "size_groups",
+    }
+
+    def __init__(
+        self,
+        title: Optional[str] = None,
+        repository: Optional[str] = None,
+        last_execution: Optional[datetime] = None,
+        total_stats: Optional[CodeCheckRunStatistics] = None,
+        prs_stats: Optional[CodeCheckRunStatistics] = None,
+        size_groups: Optional[List[int]] = None,
+    ):
+        """FilteredCodeCheckRun - a model defined in OpenAPI
+
+        :param title: The title of this FilteredCodeCheckRun.
+        :param repository: The repository of this FilteredCodeCheckRun.
+        :param last_execution: The last_execution of this FilteredCodeCheckRun.
+        :param total_stats: The total_stats of this FilteredCodeCheckRun.
+        :param prs_stats: The prs_stats of this FilteredCodeCheckRun.
+        :param size_groups: The size_groups of this FilteredCodeCheckRun.
+        """
+        self._title = title
+        self._repository = repository
+        self._last_execution = last_execution
+        self._total_stats = total_stats
+        self._prs_stats = prs_stats
+        self._size_groups = size_groups
+
+    @property
+    def title(self) -> str:
+        """Gets the title of this FilteredCodeCheckRun.
+
+        Unique name of the check run.
+
+        :return: The title of this FilteredCodeCheckRun.
+        """
+        return self._title
+
+    @title.setter
+    def title(self, title: str):
+        """Sets the title of this FilteredCodeCheckRun.
+
+        Unique name of the check run.
+
+        :param title: The title of this FilteredCodeCheckRun.
+        """
+        if title is None:
+            raise ValueError("Invalid value for `title`, must not be `None`")
+
+        self._title = title
+
+    @property
+    def repository(self) -> str:
+        """Gets the repository of this FilteredCodeCheckRun.
+
+        Repository name which uniquely identifies any repository in any service provider.
+        The format matches the repository URL without the protocol part. No \".git\" should be
+        appended. We support a special syntax for repository sets: \"{reposet id}\" adds all
```

Comment:
Are we safe wrt GHE pointers? We expect webapp to reconstruct the url of the repo.

Reply:
Yes, we should be fine here.